nautobot/extras/tests/test_jobs.py

```diff
@@ -606,6 +606,20 @@ def test_job_profiling(self):
         self.assertTrue(profiling_result.exists())
         profiling_result.unlink()
 
+    def test_job_singleton(self):
+        module = "singleton"
+        name = "TestSingletonJob"
+
+        job_class, _ = get_job_class_and_model(module, name, "local")
+        self.assertTrue(job_class.is_singleton)
+        create_job_result_and_run_job(module, name)
+        self.assertEqual(cache.get(job_class.singleton_cache_key), 1)
+        failed_job_result = create_job_result_and_run_job(module, name)
```

Comment:
I don't think this will work as a test, as `create_job_result_and_run_job` runs the requested job to completion before returning. Probably it would be sufficient as a test to simply set the cache key and then try to run the job once?

Reply:
Thank you, that explains why I was struggling to make it work.